tx-pool/src/process.rs

```diff
@@ -1029,6 +1071,8 @@ impl TxPoolService {
             check_if_hardfork_during_blocks(&hardfork_switch, &detached_blocks);
         let hardfork_during_attach =
             check_if_hardfork_during_blocks(&hardfork_switch, &attached_blocks);
+
+        let new_tip_after_delay = after_delay_window(&snapshot);
```

Comment:
I think it's OK, just mention it here.

The delayed transactions will not be processed as soon as it could.

Because this `snapshot` contains the old tip before reorg blocks.

So the delayed transactions will be processed in next reorg, not the turn when `new_tip.epoch_number == rfc_0032 && new_tip.block_index > proposal_window.farthest()`  is satisfied.

```
             hardfork-at
                 |
                 V      | no-delayed-txs
blocks ----(-----o-----)|------------|------------------------
           | delay-txs ||     N      | process-delayed-txs
               |   attached_blocks   |
                  
```

`N` is equal to or greater than `0`;  it depends on `attached_blocks`  which contains the first hardfork block, in `update_tx_pool_for_reorg`.